src/protagonist/API/Features/DeliveryChannels/Validation/HydraDeliveryChannelPolicyValidator.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.RegularExpressions;
+using FluentValidation;
+
+namespace API.Features.DeliveryChannels.Validation;
+
+/// <summary>
+/// Validator for model sent to POST /deliveryChannelPolicies and PUT/PATCH /deliveryChannelPolicies/{id}
+/// </summary>
+public class HydraDeliveryChannelPolicyValidator : AbstractValidator<DLCS.HydraModel.DeliveryChannelPolicy>
+{
+    public HydraDeliveryChannelPolicyValidator()
+    {
+        RuleFor(p => p.Id)
+            .Empty()
+            .WithMessage(p => $"DLCS must allocate named delivery channel policy id, but id {p.Id} was supplied");
+        RuleFor(p => p.CustomerId)
+            .Empty()
+            .WithMessage("Should not include user id");
+        RuleSet("post", () =>
+        {
+            RuleFor(p => p.Name)
+                .NotEmpty()
+                .WithMessage("'name' is required");
+            RuleFor(p => p.PolicyData)
```

Comment:
Could the format be validated here too? Would mean we can reject invalid policies at the boundary rather than further down, (so we do less work to say no).